### DISCUSSION TITLE
How to localize?

### DISCUSSION THREAD

**Discussion by merikonjatta** [oneduo/nova-file-manager#400] [Q&A] [answered]
Hi. I'm trying to translate the file manager's UI strings into Japanese.

I've read the instructions at https://oneduo.github.io/nova-file-manager/localization.html but
I'm not clear on how or where to define the strings.

Here are the things I've tried:
- Add keys (with Japanese translations) to `lang/ja.json`
- Add keys to `lang/vendor/nova/ja.json`
- Add keys to `lang/vendor/nova/ja.json` then append to NovaServiceProvider's boot() function: `Nova::translations(base_path('lang/vendor/nova/ja.json'));`

But none of it has worked. The UI is still displayed in English.

Any help would be appreciated.

**Comment by mikaelpopowicz** [comment, marked as answer]
Hi @merikonjatta 

You can override the `ui.php` file and put it in `lang` or `resources/lang` folder :
- `lang/vendor/nova-file-manager/ja/ui.php`
- or
- `resources/lang/vendor/nova-file-manager/ja/ui.php`

Not all labels are localized in the Vue components. We would appreciate any PR to fix that.

Cheers

**Reply by merikonjatta** [reply]
Thank you!
